Automattic/vip-go-mu-plugins
# Implement a simple async scheduler to add a cron event to be executed some time later.

## Description

Currently, we lack the ability to quickly schedule an arbitrary long-running command to run across the environments.

This PR provides this missing piece. 

NB: Cron Control is a hard dependency for this, mostly for consistency and minifying the chances of any object cache related surprises. 

### Example Usage

#### To schedule a command:

```
wp vip cmd-scheduler schedule --cmd='wp some command-to-run --with-argument' --when=now
```

`--cmd` is the command to run.

`--when` accepts either a unix timestamp or 'now'

#### To check the status of the command

```
wp vip cmd-scheduler check-status --cmd='wp some command-to-run --with-argument'
```
=> 

```
Success: Found event, status: pending; created: 2020-08-19 02:51:59; last modified: 2020-08-19 02:51:59
```

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Check out PR.
1. Schedule and check some commands as per "example usage" section.
